app/configurator/components/chart-controls/color-ramp.tsx

```diff
@@ -1,34 +1,209 @@
+        sx={{
+          color: "monochrome700",
+          display: "flex",
+          justifyContent: "space-between",
+          alignItems: "center",
+          bg: "monochrome100",
+          p: 1,
+          height: "40px",
+          borderWidth: "1px",
+          borderStyle: "solid",
+          borderColor: "monochrome500",
+          ":hover": {
+            bg: "monochrome100",
+          },
+          ":active": {
+            bg: "monochrome100",
+          },
+          ":disabled": {
+            cursor: "initial",
+            bg: "muted",
+          },
+        }}
+      >
+        {state.state === "CONFIGURING_CHART" && (
+          <>
+            <ColorRamp
+              colorInterpolator={currentPalette.interpolator}
+              nbClass={nbClass}
+            />
+            <Icon name="unfold" />
+          </>
+        )}
+      </Button>
+      <Box {...getMenuProps()} sx={{ bg: "monochrome100" }}>
+        {isOpen && (
+          <>
+            <Text as="div" variant="meta" sx={{ p: 1 }}>
+              <Trans id="controls.color.palette.diverging">Diverging</Trans>
+            </Text>
+            {divergingPalettes.map((d, i) => (
+              <PaletteRamp
+                key={`diverging-${i}`}
+                palette={d}
+                backgroundColor={
+                  i === highlightedIndex ? "monochrome200" : "monochrome100"
+                }
+                itemProps={getItemProps({ item: d, index: i })}
+                nbClass={nbClass}
+              />
+            ))}
+            <Text as="div" variant="meta" sx={{ p: 1 }}>
+              <Trans id="controls.color.palette.sequential">Sequential</Trans>
+            </Text>
+            {sequentialPalettes.map((d, i) => (
+              <PaletteRamp
+                key={`sequential-${i}`}
+                palette={d}
+                backgroundColor={
+                  i + divergingPalettes.length === highlightedIndex
+                    ? "monochrome200"
+                    : "monochrome100"
+                }
+                itemProps={getItemProps({
+                  item: d,
+                  index: i + divergingPalettes.length,
+                })}
+                nbClass={nbClass}
+              />
+            ))}
+          </>
+        )}
+      </Box>
+    </Box>
+  );
+};
+
+const PaletteRamp = (props: {
+  palette: {
+    label: string;
+    value: DivergingPaletteType | SequentialPaletteType;
+    interpolator: (t: number) => string;
+  };
+  backgroundColor: string;
+  itemProps: any;
+  nbClass?: number;
+}) => {
+  const { palette, backgroundColor, nbClass, itemProps } = props;
+
+  return (
+    <Box sx={{ p: 1, cursor: "pointer", backgroundColor }}>
+      <Box sx={{ backgroundColor }} {...itemProps}>
+        <ColorRamp
+          key={`option-${palette.value}`}
+          colorInterpolator={palette.interpolator}
+          nbClass={nbClass}
+        />
+      </Box>
+    </Box>
+  );
```

Comment:
Wondering if the PaletteRamp be added to the docs ?

Reply:
Done!